cmake/CMakeLists.txt

```
@@ -0,0 +1,112 @@
+set(LIBC_BOTTOM_HALF_HEADERS_PUBLIC_DIR ${PROJECT_SOURCE_DIR}/libc-bottom-half/headers/public)
+set(LIBC_TOP_HALF_MUSL_DIR ${PROJECT_SOURCE_DIR}/libc-top-half/musl)
+set(LIBC_TOP_HALF_MUSL_SRC_DIR ${LIBC_TOP_HALF_MUSL_DIR}/src)
+set(LIBC_BOTTOM_HALF_DIR ${PROJECT_SOURCE_DIR}/libc-bottom-half)
+set(LIBC_BOTTOM_HALF_CLOUDLIBC_SRC_DIR ${LIBC_BOTTOM_HALF_DIR}/cloudlibc/src)
+
+set(INSTALL_LIB_DIR lib/${MULTIARCH_TRIPLE})
+set(INSTALL_SHARE_DIR share/${MULTIARCH_TRIPLE})
+
+include(global-compiler-options.cmake)
+include(local-sysroot.cmake)
+include(predefined-macros.cmake)
+include(libc-imports.cmake)
+
+add_subdirectory(libc-bottom-half)
+add_subdirectory(libc-top-half)
+
+if(${MALLOC_IMPL} STREQUAL "dlmalloc")
+  add_subdirectory(dlmalloc)
+endif()
+
+add_library(wasi-emulated-signal)
+target_link_libraries(wasi-emulated-signal
+  wasi-emulated-signal-musl-objs
+  wasi-emulated-signal-objs
+)
+add_custom_command(TARGET wasi-emulated-signal POST_BUILD
+  COMMAND ${CMAKE_C_COMPILER_AR} crs $<TARGET_FILE:c>
+)
+
+add_library(c)
+
+set (LIBC_LINK_LIBRARIES
+  libc-bottom-half
+)
+
+if(${BUILD_LIBC_TOP_HALF} STREQUAL "ON")
```

I tend to prefer the more concise spelling:

```suggestion
if(BUILD_LIBC_TOP_HALF)
```

This has the benefit of allowing the user to specify the value with a truthy value that CMake permits (ON, YES, 1).